### PR TITLE
Do not use `find_package()` if boost is already included as a submodule

### PR DESCRIPTION
**Pull request by greg7mdp**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,7 +16,9 @@ IF( WIN32 )
   set(BOOST_ALL_DYN_LINK OFF) # force dynamic linking for all libraries
 ENDIF(WIN32)
 
-FIND_PACKAGE(Boost 1.67 REQUIRED COMPONENTS unit_test_framework)
+if(NOT boost_headers_SOURCE_DIR)
```
Comment (heifner) [inline CMakeLists.txt:19]: From cmake docs: `Regardless of the mode used, a <PackageName>_FOUND variable will be set to indicate whether the package was found.` I was expecting `boost_FOUND`.

**Review by heifner — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (greg7mdp) [reply CMakeLists.txt:19]: Checking for `boost_FOUND` does not work. I assume that this is set only by `find_package()`, and in our case boost is added via `add_subdirectory`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -64,10 +66,11 @@ endif()
 
 file(GLOB HEADERS "include/chainbase/*.hpp")
 add_library( chainbase src/chainbase.cpp src/pinnable_mapped_file.cpp ${HEADERS} )
-target_link_libraries( chainbase ${PLATFORM_LIBRARIES} Boost::boost)
+target_link_libraries( chainbase ${PLATFORM_LIBRARIES}
+                        Boost::headers Boost::interprocess Boost::chrono Boost::multi_index Boost::lexical_cast Boost::asio )
```
Comment (spoonincode) [inline CMakeLists.txt:70]: most of these targets don't exist in the non-cmake boost install.

Comment (greg7mdp) [reply CMakeLists.txt:70]: But they do seem needed in the boost cmake use case. Do you know of a way around it?

File: CMakeLists.txt (hunk 1 above)
Comment (greg7mdp) [reply CMakeLists.txt]: done, thanks Matt.

**Review by spoonincode — APPROVED** [review]

**Review by heifner — APPROVED** [review]

**Pull request merged**